lib/sinon/spy-entire-object.js

```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+var getPropertyDescriptor = require("./util/core/get-property-descriptor");
+var walk = require("./util/core/walk");
+
+function spyEntireObject(spy, object) {
+    walk(object || {}, function(prop, propOwner) {
```

Comment:
Maybe we could have this callback as an arrow function?

Reply:
the entire codebase is ES5.1, so arrow functions are out.